src/test/java/org/sagebionetworks/bridge/services/OAuthProviderServiceTest.java

```diff
@@ -332,4 +376,72 @@ public void introspect_ErrorCallingIntrospect() throws Exception {
         service.addScopesToAccessGrant(PROVIDER_WITH_INTROSPECT, grant);
         assertTrue(grant.getScopes().isEmpty());
     }
+    
+    @Test
+    public void oauthSignIn() throws Exception {
+        // This is not encrypted, the real token is public/private key encrypted. We mock the parser
+        // to avoid having to sign the payload.
+        mockAccessGrantCall(201, "{\"access_token\":\"not used\",\"id_token\":{\"userid\":\"12345\"}}");
+        
+        when(service.getJwtParser()).thenReturn(mockJwtParser);
+        when(mockJwtParser.parseClaimsJws(any())).thenReturn(mockJwtClaims);
+        when(mockJwtClaims.getBody()).thenReturn(mockClaims);
+        when(mockClaims.get(SYNAPSE_USERID_KEY, String.class)).thenReturn("12345");
```

Comment:
For testing purposes, let's make the "unencrypted" value different from the "encrypted" value to make sure our code is plumbing the values through correctly.

Reply:
Changed